miner/builder.go

```diff
@@ -0,0 +1,424 @@
+// Copyright 2023 The go-ethereum Authors
```

Comment:
This is a new file, but am I correct to assume that the bulk of this code has been moved over from somewhere else? 
Because if so, perhaps it would make sense to do an initial commit (or even an initial PR) which only does the non-modifying code-move, so the second commit (or PR) can focus solely on the changes. 

As it is, it's hard to evaluate "before" versus "after"

Reply:
Hmm kinda, the bulk of the code was moved from the worker. I can rename the file worker and change the copyright header so github will make the connection. Untangling the changes back would be almost impossible, theres a lot of small things that have been changed during the refactor. I think it would be best to evaluate the changes on their own without looking to much at the old code